ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-text-input/activityTextInput.component.html

```diff
@@ -3,64 +3,69 @@
+            [maxlength]="block.maxLength"
+            [pattern]="block.regexPattern"
+            [placeholder]="placeholder || block.placeholder"
+            [attr.data-ddp-test]="'answer:' + block.stableId"
+            [matAutocomplete]="autoCompleteFromSource"
+            (input)="onInput($event.target.value)"
+        />
 
-    <mat-autocomplete
-      #autoCompleteFromSource="matAutocomplete"
-      class="autoCompletePanel"
-    >
-      <mat-option
-        *ngFor="let suggestion of filteredSuggestions$ | async"
-        class="autoCompleteOption"
-        [value]="suggestion.value"
-      >
+        <mat-autocomplete
+            #autoCompleteFromSource="matAutocomplete"
+            class="autoCompletePanel"
+        >
+            <mat-option
+                *ngFor="let suggestion of filteredSuggestions$ | async"
+                class="autoCompleteOption"
+                [value]="suggestion.value"
+            >
         <span
-          class="suggestion"
-          [innerHTML]="
+            class="suggestion"
+            [innerHTML]="
             generateSuggestionOptionInnerHtml(
               suggestion,
               'span',
               'suggestion-match'
             )
           "
         ></span>
-      </mat-option>
-    </mat-autocomplete>
-  </mat-form-field>
+            </mat-option>
+        </mat-autocomplete>
+    </mat-form-field>
 
-  <ng-container *ngIf="block.confirmEntry">
-    <p
-      *ngIf="block.confirmPrompt"
-      class="ddp-question-prompt ddp-question-prompt-confirmation"
-      [class.ddp-required-question-prompt]="this.block.isRequired"
-      [innerHTML]="block.confirmPrompt"
-    ></p>
+    <ng-container *ngIf="block.confirmEntry">
+        <p
+            *ngIf="block.confirmPrompt"
+            class="ddp-question-prompt ddp-question-prompt-confirmation"
+            [class.ddp-required-question-prompt]="this.block.isRequired"
+            [innerHTML]="block.confirmPrompt"
+        ></p>
 
-    <mat-form-field class="example-input-field-confirmation">
-      <input
-        matInput
-        type="text"
-        [formControl]="formGroup.get(confirmationControlName)"
-        [placeholder]="block.confirmPlaceholder || placeholder"
-        [attr.data-ddp-test]="'answer-confirmation:' + block.stableId"
-      />
-    </mat-form-field>
-  </ng-container>
+        <mat-form-field class="example-input-field-confirmation">
+            <input
+                matInput
+                type="text"
+                [formControl]="formGroup.get(confirmationControlName)"
+                [placeholder]="block.confirmPlaceholder || placeholder"
+                [attr.data-ddp-test]="'answer-confirmation:' + block.stableId"
+            />
+        </mat-form-field>
+    </ng-container>
+    <p class="ddp-helper">
+            <span class="ddp-counter-color">
+                {{ (charactersRemaining ? charactersRemaining : block.answer) | charactersRemaining : block.maxLength }}
```

Comment:
**#Optional**
I think it could be simplified like this:
`charactersRemaining || block.answer` or depends on the case -> 
`charactersRemaining || block.answer`